akka-projection-core/src/main/scala/akka/projection/internal/ProjectionBehavior.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.internal
+
+import scala.util.Failure
+import scala.util.Success
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.InternalApi
+import akka.projection.Projection
+import akka.projection.ProjectionId
+
+@InternalApi
+private[akka] object ProjectionBehavior {
+
+  sealed trait ProjectionCommand
+  object Start extends ProjectionCommand
+  object Stop extends ProjectionCommand
+  case class RetryStop(retries: Int) extends ProjectionCommand
+  object Stopped extends ProjectionCommand
+
+  // FIXME: use stash to handle stop/start sequences
+  private[akka] def apply[Envelope](projectionFactory: () => Projection[Envelope]) = {
+
+    def started(projection: Projection[Envelope]): Behavior[ProjectionCommand] =
+      Behaviors.setup[ProjectionCommand] { ctx =>
+
+        ctx.log.info("Starting projection {}", projection.projectionId)
+        projection.run()(ctx.system)
+
+        Behaviors.receiveMessagePartial {
+          case Stop =>
+            val stoppedFut = projection.stop()(ctx.executionContext)
+            ctx.pipeToSelf(stoppedFut) {
+              case Success(_) => Stopped
+              // FIXME: we may retry to stop a few times before letting it crash
+              case Failure(ex) => throw ex
```

Comment:
Failure isn't a problem here, I think. It means that the projection stream has stopped.

Reply:
you mean, we just ignore it and send the Stopped?

Reply:
yes, I think that is the right thing